File: staff-ui/src/types/invitation.ts
export interface InvitationResponse {
  id: string;
  staffMemberId: string;
  email: string;
  status: 'Pending' | 'Accepted' | 'Expired' | 'Cancelled';
  expiresAt: string;
  createdAt: string;
  acceptedAt: string | null;
  isExpired: boolean;
}

export interface AcceptInvitationRequest {
  token: string;
  password: string;
  firstName?: string;
  lastName?: string;
}

File: staff-api/staff-domain/Entities/Invitation.cs
using System.ComponentModel.DataAnnotations;
using staff_domain.Enums;

namespace staff_domain.Entities;

public class Invitation
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid StaffMemberId { get; set; }

    [Required]
    [MaxLength(128)]
    public string TokenHash { get; set; } = string.Empty;

    [Required]
    [MaxLength(255)]
    public string Email { get; set; } = string.Empty;

    [Required]
    public DateTime ExpiresAt { get; set; }

    public InvitationStatus Status { get; set; } = InvitationStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? AcceptedAt { get; set; }

    // Navigation property
    public StaffMember StaffMember { get; set; } = null!;
}
